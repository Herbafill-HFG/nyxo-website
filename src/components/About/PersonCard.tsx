import { Link } from "gatsby"
import Image, { FluidObject } from "gatsby-image"
import React, { FC } from "react"
import styled from "styled-components"
import { device } from "../Primitives"

type Props = {
  position: string
  name: string
  image: FluidObject
  slug: string
}

const PersonCard: FC<Props> = ({ image, name, position, slug }) => {
  return (
    <Person to={`/author/${slug}`}>
      <ImageWrap>
        <Image alt={name} fluid={image} />
      </ImageWrap>
      <InfoWrap>
        <Name>{name}</Name>
        <Row>
          <Position>{position}</Position>
        </Row>
      </InfoWrap>
    </Person>
  )
}

export default PersonCard

const Person = styled(Link)`
  display: flex;
  flex-direction: row;
  margin: 1rem 0.5rem;
  padding: 1rem 0.5rem;
  border-radius: 5px;
  box-shadow: var(--shadow);
  box-sizing: border-box;
  transition: transform 0.15s ease-in;
  &:hover {
    transform: translateY(-2px);
  }

  @media ${device.desktopL} {
    max-width: calc(25% - 2 * 0.5rem);
    flex: 1 1 calc(25% - 2 * 0.5rem);
  }

  @media ${device.desktop} {
    max-width: calc(33.3333% - 2 * 0.5rem);
    flex: 1 1 calc(33.3333% - 2 * 0.5rem);
  }

  @media ${device.laptopL} {
    max-width: calc(33.3333% - 2 * 0.5rem);
    flex: 1 1 calc(33.3333% - 2 * 0.5rem);
  }

  @media ${device.laptop} {
    max-width: calc(50% - 2 * 0.5rem);
    flex: 1 1 calc(50% - 2 * 0.5rem);
  }

  @media ${device.tabletL} {
    max-width: calc(50% - 2 * 0.5rem);
    flex: 1 1 calc(50% - 2 * 0.5rem);
  }

  @media ${device.tablet} {
    max-width: calc(50% - 2 * 0.5rem);
    flex: 1 1 calc(50% - 2 * 0.5rem);
  }

  @media ${device.mobileL} {
    max-width: 100%;
    flex: 1 1 100%;
  }

  @media ${device.mobileM} {
    max-width: 100%;
    flex: 1 1 100%;
  }

  @media ${device.mobileS} {
    max-width: 100%;
    flex: 1 1 100%;
  }
`

const ImageWrap = styled.div`
  width: 3rem;
  height: 3rem;
  min-width: 50px;

  img {
    border-radius: 50%;
  }
`

const InfoWrap = styled.div`
  margin-left: 30px;
`

const Position = styled.div`
  font-size: 0.9rem;
  font-family: var(--medium);
  color: var(--textSecondary);
`

const Row = styled.div`
  display: flex;
  flex-direction: row;
`

const Name = styled.h3`
  margin: 0rem 0rem 0.5rem;
`
