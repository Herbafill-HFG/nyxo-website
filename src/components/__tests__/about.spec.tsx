import React from "react"
import renderer from "react-test-renderer"

import PersonCard from "../About/PersonCard"

describe("PersonCard", () => {
  const position = "Some title"
  const name = "1234556789"
  const image = "/static/12427087eb4503fdd2e65051f774db4e/47813/anu.jpg"

  it("renders correctly", () => {
    const tree = renderer
      .create(<PersonCard name={name} position={position} image={image} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
