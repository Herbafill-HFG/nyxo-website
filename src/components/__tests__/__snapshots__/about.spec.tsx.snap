// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard renders correctly 1`] = `
<a
  className="sc-AxmLO dLyVYA"
  href="/author/undefined"
>
  <div
    className="sc-fzozJi geTWwZ"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "maxHeight": null,
          "maxWidth": null,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "NaN%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source />
        <img
          alt="1234556789"
          loading="lazy"
          onLoad={[Function]}
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"1234556789\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-fzoLsD jezVAk"
  >
    <h3
      className="sc-fznyAO hlQiAB"
    >
      1234556789
    </h3>
    <div
      className="sc-fzplWN ckjgcB"
    >
      <div
        className="sc-fzpans dtPbUO"
      >
        Some title
      </div>
    </div>
  </div>
</a>
`;
